### ISSUE TITLE
`redundant_closure` suggestion ignores capture by reference

### ISSUE THREAD

**Issue by SamRodri** [rust-lang/rust-clippy#8063]
The suggestion for `redundant_closure` lint ignores capture by reference, causing compile error if the referenced closure is reused.

I tried this code:

```rust
pub fn test(bar: impl Fn()) {
    foo(|| bar());
    bar();
}
pub fn foo(_bar: impl FnOnce()) { }
```

The suggestion `foo(bar)` causes the closure to move so it cannot be reused after, I expected either `foo(&bar)` suggestion or no warning.

### Meta

Rust version (`rustc -Vv`)

```
rustc 1.57.0 (f1edd0429 2021-11-29)
binary: rustc
commit-hash: f1edd0429582dd29cccacaf50fd134b05593bd9c
commit-date: 2021-11-29
host: x86_64-pc-windows-msvc
release: 1.57.0
LLVM version: 13.0.0
```

@rustbot label +I-suggestion-causes-error

**Comment by SamRodri** [comment]
The suggestion is now `foo(&mut bar)`, still incorrect, but less confusing.

```
rustc 1.68.1 (8460ca823 2023-03-20)
binary: rustc
commit-hash: 8460ca823e8367a30dda430efda790588b8c84d3
commit-date: 2023-03-20
host: x86_64-pc-windows-msvc
release: 1.68.1
LLVM version: 15.0.6
```

**Comment by SamRodri** [comment]
Tested this again with `clippy 0.1.83 (90b35a6 2024-11-26)`, the issue is fixed. Not sure when it was fixed.
